app/src/main/java/org/wikipedia/WikipediaApp.kt

```diff
@@ -239,21 +240,24 @@ class WikipediaApp : Application() {
         defaultWikiSite = null
     }
 
-    @SuppressLint("CheckResult")
     fun logOut() {
-        L.d("Logging out")
-        AccountUtil.removeAccount()
-        Prefs.isPushNotificationTokenSubscribed = false
-        Prefs.pushNotificationTokenOld = ""
-        ServiceFactory.get(wikiSite).getTokenObservable()
-                .subscribeOn(Schedulers.io())
-                .flatMap {
-                    val csrfToken = it.query!!.csrfToken()
-                    WikipediaFirebaseMessagingService.unsubscribePushToken(csrfToken!!, Prefs.pushNotificationToken)
-                            .flatMap { ServiceFactory.get(wikiSite).postLogout(csrfToken).subscribeOn(Schedulers.io()) }
-                }
-                .doFinally { SharedPreferenceCookieManager.instance.clearAllCookies() }
-                .subscribe({ L.d("Logout complete.") }) { L.e(it) }
+        MainScope().launch(CoroutineExceptionHandler { _, t ->
+            L.e(t)
+        }) {
+            try {
+                L.d("Logging out")
+                AccountUtil.removeAccount()
+                Prefs.isPushNotificationTokenSubscribed = false
+                Prefs.pushNotificationTokenOld = ""
+
+                val token = ServiceFactory.get(wikiSite).getToken().query!!.csrfToken()
+                WikipediaFirebaseMessagingService.unsubscribePushToken(token!!, Prefs.pushNotificationToken)
+                ServiceFactory.get(wikiSite).postLogout(token)
+            } finally {
+                SharedPreferenceCookieManager.instance.clearAllCookies()
+                L.d("Logout complete.")
+            }
+        }
```

Comment:
Have another idea on this instead `try-finally` inside the `launch` block:

Adding `.invokeOnCompletion { }` and treat it as a `finally`:

```
MainScope().launch(CoroutineExceptionHandler { _, t ->
            L.e(t)
}) {
...
}.invokeOnCompletion {
   SharedPreferenceCookieManager.instance.clearAllCookies()
   L.d("Logout complete.")
}
```

Reply:
Sure!